cli/azd/pkg/input/console.go

```diff
@@ -63,14 +67,35 @@ func (c *AskerConsole) SetWriter(writer io.Writer) {
 
 // Prints out a message to the underlying console write
 func (c *AskerConsole) Message(ctx context.Context, message string) {
-	// Only write to the console during interactive & non-formatted responses.
-	if c.interactive && (c.formatter == nil || c.formatter.Kind() == output.NoneFormat) {
+	if c.formatter != nil && c.formatter.Kind() == output.JsonFormat {
+		// we do not expect formatter.Format to fail on valid input. Discard the error to appease the linter (note
+		// that we implicitly discard the error from fmt.Fprintln later, as well).
+		_ = c.formatter.Format(c.consoleMessageForMessage(message), c.writer, nil)
```

Comment:
I'm thinking through this more, and I wonder if formatted messages output is part of a solution to solve "asynchronous long-running operations" rather than the general azd contract.

For example, what do you think something like `azd env get-values --output json` should output?
`{ "type": "envOutputJson", "data": { "env_key_1": "env_value" }}` 

or simply `{"env_key_1": "env_value"}`?

Reply:
This is a good question - I realized that we also have the problem today with `azd template list --output=json`.

I think we want to make this stuff regular for all commands, and so if we are going to use an envelope, the output should be:

```
{ "type": "envOutputJson", "data": { "env_key_1": "env_value" }}
```

I agree this feels a little worse than our current solution, but I'm not sure the best way for us to solve this problem.  I think if we are going to mix multiple objects over stdout we need some way for consumers to figure out how to decode them and so they need some type tags.

The other option would be to say when using `--output=json` we write the final value (the "command result") to stdout and we use stderr as the "event stream" and these JSON objects do have envelopes since we imagine emitting multiple types of data long term?

Reply:
The concern has been addressed. We decided on `stdout` for result output, `stderr` for event stream.